### PR TITLE
add cpp solution + editorial for circlecross (gold)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
o.o

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-719.mdx [hunk 1]
```diff
@@ -2,17 +2,118 @@
...
 <LanguageSection>
-  
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <vector>
+#include <map>
+
```
Comment (jessechoe10) [inline solutions/usaco-719.mdx:30]: just do ```using namespace std;```

**Review by jessechoe10 — CHANGES_REQUESTED** [review]

File: solutions/usaco-719.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/usaco-719.mdx:30]: no

File: solutions/usaco-719.mdx [hunk 2]
```diff
@@ -2,17 +2,118 @@
...
+		std::cin >> cow_id;
+
+		// zero-index the cows
+		loc[i] = --cow_id;
+	}
+
+	Seg<int> cross;
+	cross.init(2 * n);
+
+	int ans = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-719.mdx:93]: more meaningful name

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-719.mdx [hunk 3]
```diff
@@ -2,17 +2,118 @@
...
+```cpp
+#include <iostream>
+#include <cstdio>
+#include <vector>
+#include <map>
+
+using std::vector;
+
+//BeginCodeSnip{Segment Tree Implementation}
+template<class T> struct Seg {
```
[the remark this answers is not in the document]
Comment (ryanchou-dev) [reply solutions/usaco-719.mdx:34]: this is the implementation mentioned in the module, so i think ill leave it like that

File: solutions/usaco-719.mdx [hunk 4]
```diff
@@ -2,17 +2,118 @@
...
+	for (int i = 0; i < 2 * n; i++) {
+		int cow_id;
+		std::cin >> cow_id;
+
+		// zero-index the cows
+		loc[i] = --cow_id;
+	}
+
+	Seg<int> cross;
+	cross.init(2 * n);
```
[the remark this answers is not in the document]
Comment (ryanchou-dev) [reply solutions/usaco-719.mdx:91]: ^^

File: solutions/usaco-719.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (SansPapyrus683) [reply solutions/usaco-719.mdx:34]: fair enough

File: solutions/usaco-719.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/usaco-719.mdx:93]: no like for `ans` lol

Comment (ryanchou-dev) [reply solutions/usaco-719.mdx:93]: oops ok

**Review by SansPapyrus683 — APPROVED** [review]
bing chilling

**Review by TruaShamu — APPROVED** [review]
looks good to me :)

**Review by RedBlueBird — APPROVED** [review]
Perfection!
The only comment I have is `using namespace std;` but apparently sans doesn't like that XD

**Pull request merged**